controllers/sriovnetworknodeconfig_controller.go

```diff
@@ -0,0 +1,183 @@
+			// 1. MachineConfigPool name matches with OvsHardwareOffloadConfig.Name
+			// 2. MachineConfigPool nodeSelector label(s) matches with OvsHardwareOffloadConfig.NodeSelector
+			if mcp.GetName() == nodeConfig.Spec.OvsHardwareOffloadConfig.Name {
+				if mcp.GetName() == "master" {
+					logger.Info("Master nodes are selected viby OvsHardwareOffloadConfig.Name, ignoring.")
+					continue
+				}
+				mcpMap[mcp.GetName()] = true
+				break
+			}
+			selector, err := metav1.LabelSelectorAsSelector(mcp.Spec.NodeSelector)
+			if err != nil {
+				return fmt.Errorf("Invalid label selector in MachineConfigPool: %s, %v", mcp.GetName(), err)
+			}
+			if selector.Matches(labels.Set(nodeConfig.Spec.OvsHardwareOffloadConfig.NodeSelector)) {
+				// OVS Hardware Offload is not supported on master nodes
+				if mcp.GetName() == "master" {
+					logger.Info("Master nodes are selected by OvsHardwareOffloadConfig.NodeSelector, ignoring.")
+					continue
+				}
+				mcpMap[mcp.GetName()] = true
+				break
+			}
+		}
+	}
+
+	for mcpName, enable := range mcpMap {
+		mcName := "00-" + mcpName + "-" + OVS_HWOL_MACHINE_CONFIG_NAME_SUFFIX
+		mc, err := render.GenerateMachineConfig("bindata/manifests/switchdev-config", mcName, mcpName, true, &data)
+		if err != nil {
+			return err
+		}
+
+		foundMC := &mcfgv1.MachineConfig{}
+		err = r.Get(context.TODO(), types.NamespacedName{Name: mcName}, foundMC)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				if enable {
+					err = r.Create(context.TODO(), mc)
+					if err != nil {
+						return fmt.Errorf("Couldn't create MachineConfig: %v", err)
+					}
+					logger.Info("Created MachineConfig CR in MachineConfigPool", mcName, mcpName)
+				}
+			} else {
+				return fmt.Errorf("Failed to get MachineConfig: %v", err)
+			}
+		} else {
+			if enable {
+				if bytes.Compare(foundMC.Spec.Config.Raw, mc.Spec.Config.Raw) == 0 {
+					logger.Info("MachineConfig already exists, updating")
+					err = r.Update(context.TODO(), foundMC)
+					if err != nil {
+						return fmt.Errorf("Couldn't update MachineConfig: %v", err)
+					}
+				} else {
+					logger.Info("No content change, skip updating MC")
+				}
+			} else {
+				logger.Info("offload disabled, delete MachineConfig")
+				err = r.Delete(context.TODO(), foundMC)
+				if err != nil {
+					return fmt.Errorf("Couldn't delete MachineConfig: %v", err)
+				}
+			}
+		}
+	}
+
+	// Remove stale MCs for MCP that no longer exists in OvsHardwareOffloadConfig
+	for _, mcp := range mcpList.Items {
+		found := false
+		for mcpName := range mcpMap {
+			if mcp.Name == mcpName {
+				found = true
+				break
+			}
+		}
+		if !found {
+			mcName := "00-" + mcp.Name + "-" + OVS_HWOL_MACHINE_CONFIG_NAME_SUFFIX
+			foundMC := &mcfgv1.MachineConfig{}
+			err = r.Get(context.TODO(), types.NamespacedName{Name: mcName}, foundMC)
+			if err != nil {
+				if !errors.IsNotFound(err) {
+					err = r.Delete(context.TODO(), foundMC)
```

Comment:
This shall be invoked when the MC CR is found without error.  It shall be moved to line 177.

Reply:
Fixed